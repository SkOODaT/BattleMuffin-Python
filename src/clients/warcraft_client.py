import requests
from authlib.integrations.requests_client import OAuth2Session
from uplink import Consumer, Query, get, params, returns

from config.client_config import ClientConfiguration
from config.region_config import Locale, Region


class WarcraftClient(Consumer):
    def __init__(self, client_id: str, client_secret: str, region: Region = Region.us, locale: Locale = Locale.en_US):
        self.client_config = ClientConfiguration(client_id, client_secret, region, locale)

        oath_response = requests.get(f"{self.client_config.oauth_host}/oauth/.well-known/openid-configuration").json()
        token_endpoint = oath_response['token_endpoint']
        session = OAuth2Session(client_id, client_secret, scope="openid", token_endpoint=token_endpoint, grant_type="client_credentials")
        session.fetch_token()
        super().__init__(base_url=self.client_config.host, client=session)
        self._inject(Query("locale").with_value(self.client_config.locale.name))

    @returns.json
    @params({"namespace": "static-us"})
    @get("/data/wow/achievement-category/index")
    def get_achievement_category_index(self):
        """Returns an index of achievement categories."""
