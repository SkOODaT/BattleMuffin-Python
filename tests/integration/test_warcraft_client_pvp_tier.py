import os
from clients.warcraft_client import WarcraftClient


def test_get_pvp_tiers_index(snapshot):
    client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
    response = client.get_pvp_tiers_index()
    assert response == snapshot


def test_get_pvp_tier(snapshot):
    client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
    response = client.get_pvp_tier(1)
    assert response == snapshot


def test_get_pvp_tier_media(snapshot):
    client = WarcraftClient(os.getenv("CLIENT_ID"), os.getenv("CLIENT_SECRET"))
    response = client.get_pvp_tier_media(1)
    assert response == snapshot
