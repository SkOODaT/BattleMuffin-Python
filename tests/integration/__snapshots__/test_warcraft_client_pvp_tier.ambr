# name: test_get_pvp_tier
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/1?namespace=static-8.3.0_32861-us',
      },
    },
    'bracket': <class 'dict'> {
      'id': 0,
      'type': 'ARENA_2v2',
    },
    'id': 1,
    'max_rating': 1400,
    'media': <class 'dict'> {
      'id': 1,
      'key': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/media/pvp-tier/1?namespace=static-8.3.0_32861-us',
      },
    },
    'min_rating': 0,
    'name': 'Unranked',
    'rating_type': 0,
  }
---
# name: test_get_pvp_tier_media
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/media/pvp-tier/1?namespace=static-8.3.0_32861-us',
      },
    },
    'assets': <class 'list'> [
      <class 'dict'> {
        'key': 'icon',
        'value': 'https://render-us.worldofwarcraft.com/icons/56/ui_rankedpvp_01.jpg',
      },
    ],
    'id': 1,
  }
---
# name: test_get_pvp_tiers_index
  <class 'dict'> {
    '_links': <class 'dict'> {
      'self': <class 'dict'> {
        'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/?namespace=static-8.3.0_32861-us',
      },
    },
    'tiers': <class 'list'> [
      <class 'dict'> {
        'id': 1,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/1?namespace=static-8.3.0_32861-us',
        },
        'name': 'Unranked',
      },
      <class 'dict'> {
        'id': 2,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/2?namespace=static-8.3.0_32861-us',
        },
        'name': 'Combatant',
      },
      <class 'dict'> {
        'id': 3,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/3?namespace=static-8.3.0_32861-us',
        },
        'name': 'Challenger',
      },
      <class 'dict'> {
        'id': 4,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/4?namespace=static-8.3.0_32861-us',
        },
        'name': 'Rival',
      },
      <class 'dict'> {
        'id': 5,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/5?namespace=static-8.3.0_32861-us',
        },
        'name': 'Duelist',
      },
      <class 'dict'> {
        'id': 6,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/6?namespace=static-8.3.0_32861-us',
        },
        'name': 'Elite',
      },
      <class 'dict'> {
        'id': 8,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/8?namespace=static-8.3.0_32861-us',
        },
        'name': 'Unranked',
      },
      <class 'dict'> {
        'id': 9,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/9?namespace=static-8.3.0_32861-us',
        },
        'name': 'Combatant',
      },
      <class 'dict'> {
        'id': 11,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/11?namespace=static-8.3.0_32861-us',
        },
        'name': 'Challenger',
      },
      <class 'dict'> {
        'id': 12,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/12?namespace=static-8.3.0_32861-us',
        },
        'name': 'Rival',
      },
      <class 'dict'> {
        'id': 13,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/13?namespace=static-8.3.0_32861-us',
        },
        'name': 'Duelist',
      },
      <class 'dict'> {
        'id': 14,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/14?namespace=static-8.3.0_32861-us',
        },
        'name': 'Elite',
      },
      <class 'dict'> {
        'id': 16,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/16?namespace=static-8.3.0_32861-us',
        },
        'name': 'Unranked',
      },
      <class 'dict'> {
        'id': 17,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/17?namespace=static-8.3.0_32861-us',
        },
        'name': 'Combatant',
      },
      <class 'dict'> {
        'id': 18,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/18?namespace=static-8.3.0_32861-us',
        },
        'name': 'Challenger',
      },
      <class 'dict'> {
        'id': 19,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/19?namespace=static-8.3.0_32861-us',
        },
        'name': 'Rival',
      },
      <class 'dict'> {
        'id': 20,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/20?namespace=static-8.3.0_32861-us',
        },
        'name': 'Duelist',
      },
      <class 'dict'> {
        'id': 21,
        'key': <class 'dict'> {
          'href': 'https://us.api.blizzard.com/data/wow/pvp-tier/21?namespace=static-8.3.0_32861-us',
        },
        'name': 'Elite',
      },
    ],
  }
---
